prime-router/docs/api/permissions.yml

```diff
@@ -0,0 +1,163 @@
+openapi: "3.0.3"
+info:
+  title: ReportStream Permissions API
+  version: "1.0"
+  contact:
+    name: USDS at Centers for Disease Control and Prevention
+    url: https://open.cdc.gov
+    email: usds@cdc.gov
+paths:
+  /api/permissions:
+    get:
+      description: fetch a list of permissions
+      security:
+        - OAuth2: [ system_admin ]
+      responses:
+        '200':
+          description: a list of permission objects
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Permission'
+        '400':
+          description: bad request
+        '401':
+          description: unauthorized
+  /api/organization/{orgName}/permissions:
+    get:
+      description: fetch a list of permissions for an organization
+      security:
+        - OAuth2: [ user ]
+      parameters:
+        - name: orgName
+          in: path
+          description: a valid organization name
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: a list of permission objects
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Permission'
+        '400':
+          description: bad request
+        '401':
+          description: unauthorized
+  /api/permission/{id}:
+    put:
+      description: updates a permission
+      security:
+        - OAuth2: [ system_admin ]
+      parameters:
+        - name: id
+          in: path
+          description: a valid permissions id
+          required: true
+          schema:
+            type: integer
+      responses:
+        '400':
+          description: bad request
+        '401':
+          description: unauthorized
+    delete:
+      description: deletes a permission
+      security:
+        - OAuth2: [ system_admin ]
+      parameters:
+        - name: id
+          in: path
+          description: a valid permissions id
+          required: true
+          schema:
+            type: integer
+      responses:
+        '400':
+          description: bad request
+        '401':
+          description: unauthorized
+  /api/permission:
```

Comment:
I'm a bit confused on some of these endpoints. Using this one as an example, shouldn't we allow the caller to set different permission parameters, such as name and description, for example?

The #/components/schemas/Permission object is only referenced in the responses?

Reply:
If you look at reports.yml, it allows the client to pass in query parameters in its post, which is what I would expect to see here?

Reply:
yup, my bad - will update

Reply:
This endpoint can allow for updating all fields but our use case is only for enabled currently and thus why it can be confusing. I can make it so that it accepts the other fields but lets discuss.